src/python/tests/seq_region/test_collection.py

```diff
@@ -0,0 +1,366 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.seq_region.collection` module."""
+
+import json
+from contextlib import nullcontext as no_raise
+from pathlib import Path
+from typing import Any, ContextManager
+from unittest.mock import MagicMock, patch
+
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.Seq import Seq
+import pytest
+from pytest import param, raises
+
+from ensembl.io.genomio.seq_region.exceptions import UnknownMetadata
+from ensembl.io.genomio.seq_region.gbff import GBFFRecord
+from ensembl.io.genomio.seq_region.collection import SeqCollection
+
+
+_test_report_seq = {
+    "Assembly-Unit": "Primary Assembly",
+    "Assigned-Molecule": "I",
+    "Assigned-Molecule-Location/Type": "Chromosome",
+    "GenBank-Accn": "gb_id",
+    "RefSeq-Accn": "rs_id",
+    "Relationship": "=",
+    "Sequence-Length": "1000",
+    "Sequence-Name": "seq_name",
+    "Sequence-Role": "assembled-molecule",
+    "UCSC-style-name": "na",
+}
+
+
+class MockResponse:
+    """Mock `requests` response."""
+
+    def __init__(self, json_str: str) -> None:
+        self.text = json_str
+
+    @staticmethod
+    def raise_for_status():
+        """Mock, never raise Exception here."""
+
+    def json(self) -> dict:
+        """Returns the data as json."""
+        return json.loads(self.text)
```

Comment:
I remember seeing this in other tests. Should we move this to the `conftest.py` file to make it "global"? That being said, haven't checked if the code is exactly the same in every case.

Reply:
Done, I've made it into a new fixture `mock_response`.

Reply:
Off it goes then!
```suggestion
```